src/storage.ts

```diff
@@ -187,6 +188,16 @@ export interface StoredBranchComparisons {
 	[id: string]: string | StoredBranchComparison;
 }
 
+export interface StoredDeepLinkContext {
```

Comment:
What is the absolute minimum we can store here? The less we store the less chance we need to do data migrations. I feel like we could probably get away with the `url` only, but we probably need `state` to pick up where we left off (otherwise we have to start processing over from the url -- which probably would work in our current use-cases).

Reply:
I set this up to be flexible enough to allow us to pick up where we left off at any point, though it may be that we really only need to "store" in the "open repo" case.

1.  If we don't mind starting over completely, we probably only need the `url`, and then we can separate out the "on did receive uri" event logic and just call it on the uri again.
2.  If we want to pick up exactly where we left off, but wanted to make it specific to this case only for now, we would want at least the url, repo id and remote url, and we would transition it into the `Added Repo Match` state when we call `processDeepLink` by setting state and action accordingly.

Reply:
Actually, it looks like we might be able to just use the `url`, set the state and action to take us to "added repo match" and process the other context using our "on did receive uri" logic, separated into its own function. That might be sufficient for now.

Reply:
Updated in commit: [Improvements from feedback](https://github.com/gitkraken/vscode-gitlens/pull/2503/commits/9e01ae826ec49b794057aa9570ec3df54ef0042d)

- We store the deep link url and the repo path.
- When the storage value is read, we try to get the repo from the repo path. If we do get it, we skip straight to remote matching. If not, we process the link as normal (i.e. wait for repo discovery to add the repo)